features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/SessionReplayFeature.kt

```diff
@@ -74,7 +78,10 @@ internal class SessionReplayFeature(
     )
 
     private lateinit var appContext: Context
+    private var shouldRecord = AtomicBoolean(false)
+    private var recordingStateChanged = AtomicBoolean(false)
```

Comment:
nit: Given the complexity of the recorder logic, it might be helpful to add comments explaining the purpose of the `shouldRecord` and `recordingStateChanged` variables and how they are used throughout the class. This could make the code easier to understand and maintain.

Reply:
I've added some comments